core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -351,7 +354,35 @@ public void invalidateTable(SessionContext context, TableIdentifier ident) {}
   @Override
```

Comment:
Just wondering why can't we remove this method and let it use the parent implementation.
Other catalogs do the same.

Reply:
oh.. This is `SessionCatalog`, not the `Catalog`. So, there is no parent implementation. 
I think this logic can be moved to `BaseSessionCatalog`. So, it can help if there are some more implementations in the future.

Reply:
Thanks, @ajantha-bhat for the input. Yes, moving implementation logic makes sense to `BaseSessionCatalog`. I will update the code.

Reply:
I see that all the method's implementation is in `RESTSessionCatalog.java`. IMO we should keep `registerTable` implementation in `RESTSessionCatalog.java` as well. WDYT?